Fix const error when compiling in clang.

cianciosa opened the pull request:
## Proposed changes

This fixes the second bug in https://github.com/project-asgard/asgard/issues/387. The bug arrises because `plan_` contains elements that are marked `const` and the line `plan_ = new_plan` tries to overwrite them which would violate the `const` property. This pr works around that by deleting the old elements of `plan_` first then inserting the elements of `new_plan`

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [x] Bugfix
- [ ] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## What systems has this change been tested on?
Mac OS and Linux. Running the unit tests locally result in several test failures of unknown origin.



## Checklist

- [x] this PR is up to date with current the current state of 'develop'
- [ ] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Trigger Tests

cianciosa committed:
Revert changes to adapt.cpp

cianciosa committed:
Revert changes to adapt.cpp

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert test code that was accidently pushed.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Revert unintented change to assignment.

cianciosa committed:
Add initial function to pull git repo using cmake.

cianciosa committed:
Disable the kronmult submodule.

cianciosa committed:
Update default git URL for kronmult.

cianciosa committed:
Add custom url config for fetch content.

cianciosa committed:
Comment out all the old kronmult contib code.

cianciosa committed:
kronmult_cuda uses the build_info.hpp header link kron as public.

cianciosa committed:
Change registered kronmult project to a specific commit. Updates to this default should proceed through a pull request.

cianciosa committed:
Remove commented out code. Remove pull target. External projects do not need to be working repository.

cianciosa committed:
Force the USE_GPU kronmult option on when ASGARD_USE_CUDA is set.

cianciosa committed:
Bump kronmult tag to latest version to avoid running the sanitizers in the ci tests.

cianciosa committed:
Merge branch 'develop' of https://github.com/cianciosa/asgard into develop

cianciosa committed:
Fix const error on clang by clearing and inserting the new_plan.

cianciosa committed:
parent 8e83f8804d2f6b318794ffae53d43ee86b8983b5
author Mark Cianciosa <cianciosamr@ornl.gov> 1626113044 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626725824 -0400

parent 8e83f8804d2f6b318794ffae53d43ee86b8983b5
author Mark Cianciosa <cianciosamr@ornl.gov> 1626113044 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626725810 -0400

parent 8e83f8804d2f6b318794ffae53d43ee86b8983b5
author Mark Cianciosa <cianciosamr@ornl.gov> 1626113044 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626725808 -0400

Fix const error on clang by clearing and inserting the new_plan.

Revert test code that was accidently pushed.

Try reseting personal access token.

kronmult_cuda uses the build_info.hpp header link kron as public.

Force the USE_GPU kronmult option on when ASGARD_USE_CUDA is set.

Remove the submodule config.

Add comment explaining why new_plan is not just asigned to plan_.

quantumsteve commented on src/adapt.cpp:
```diff
@@ -312,7 +312,8 @@ fk::vector<P> distributed_grid<P>::refine_elements(
   auto const new_plan = get_plan(get_num_ranks(), table_);
   auto const remapper = remap_for_addtl(table_.size() - added);
   auto const y        = redistribute_vector(x, plan_, new_plan, remapper);
-  plan_               = new_plan;
```
Please add a comment about why we're doing this.

quantumsteve commented on src/adapt.cpp:
```diff
@@ -338,7 +339,9 @@ fk::vector<P> distributed_grid<P>::remove_elements(
   auto const remapper = remap_for_delete(all_remove_indices, table_.size());
   auto const y        = redistribute_vector(x, plan_, new_plan, remapper);
-  plan_               = new_plan;
+
```
Please add a comment about why we're doing this.

quantumsteve reviewed (changes requested):
Please add a code comment and squash your commits.

cianciosa committed:
Fix merge conflict after commit squash

cianciosa committed:
parent 6bb4ac67d22e0f1ed02c7feb0bac38887f3f3fb8
parent 35e54f54c32c97f9bbd8f5194f96c1bef5a6daf8
author cianciosa <cianciosamr@ornl.gov> 1626726001 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626726817 -0400

Fix merge conflict after commit squash

Revert test code that was accidently pushed.

parent 8e83f8804d2f6b318794ffae53d43ee86b8983b5
author Mark Cianciosa <cianciosamr@ornl.gov> 1626113044 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626725824 -0400

parent 8e83f8804d2f6b318794ffae53d43ee86b8983b5
author Mark Cianciosa <cianciosamr@ornl.gov> 1626113044 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626725810 -0400

parent 8e83f8804d2f6b318794ffae53d43ee86b8983b5
author Mark Cianciosa <cianciosamr@ornl.gov> 1626113044 -0400
committer cianciosa <cianciosamr@ornl.gov> 1626725808 -0400

Fix const error on clang by clearing and inserting the new_plan.

Revert test code that was accidently pushed.

Try reseting personal access token.

kronmult_cuda uses the build_info.hpp header link kron as public.

Force the USE_GPU kronmult option on when ASGARD_USE_CUDA is set.

Remove the submodule config.

Add comment explaining why new_plan is not just asigned to plan_.

cianciosa committed:
Merge branch 'develop' of https://github.com/cianciosa/asgard into develop

cianciosa commented:
🤬🤬🤬 Every time I try to squash the commits, it just keeps adding more.
